### PR TITLE
refactor: Standardize test method naming convention (#1427)

### LINKED ISSUE #1427: [Code health] Remove "test" prefix from unit and instrumented test functions
We currently do both - some tests are prefixed with `test` while others aren't. 

For consistency, I suggest we are consistent with implicit convention in Android testing docs:
https://developer.android.com/guide

@scolsen @shobhitagarwal1612 @JSunde Any objections?

### PR DESCRIPTION
**Pull request by AnuragManu**
…ons (#1427)

This commit addresses issue #1427 by removing the 'test' prefix from test functions and converting them to backtick style descriptive names.

- Converted camelCase test names to space-separated lowercase words
- Wrapped test names in backticks for better readability
- Reformatted test names with conditions to use 'when' clauses
- Fixed edge cases with duplicate 'when' keywords

This change improves code readability and follows the implicit convention in Android testing documentation.

Fixes #1427 

# Refactor: Standardize Test Method Naming Convention

## Overview
This PR addresses issue #1427 by standardizing the test method naming convention across the codebase. It removes the "test" prefix from unit and instrumented test functions and converts them to the backtick style naming convention, which improves readability and follows modern Kotlin testing practices.

## Changes
- Removed "test" prefix from all test functions in the codebase
- Converted test method names to backtick style descriptive names following these patterns:
  - Regular tests: `` @Test fun `descriptive test name`() ``
  - Tests with conditions: `` @Test fun `main test part when condition part`() ``
- Maintained test functionality while improving code readability
- Ensured consistent naming patterns across all test files

### Before:
```kotlin
@Test
fun testHeader() {
  // Test code
}

@Test
fun testActionButtons_whenTaskIsOptional() {
  // Test code
}
```

### After:
```kotlin
@Test
fun `header`() {
  // Test code
}

@Test
fun `action buttons when task is optional`() {
  // Test code
}
```

## Implementation Details
The changes were implemented systematically using a custom Python script that:
1. Identifies all test methods with "test" prefix
2. Converts camelCase names to space-separated lowercase words
3. Handles special cases with underscores by converting them to "when" clauses
4. Wraps the names in backticks for improved readability
5. Fixes edge cases like duplicate "when" keywords

## Testing
- Verified all unit tests pass after the naming convention changes
- Manually reviewed test files to ensure proper conversion
- Fixed syntax issues in edge cases where automated conversion needed adjustment
- Confirmed test discovery still works correctly with the new naming convention
- Validated test reports display the new names correctly

## Metrics
- Files modified: 27
- Test methods renamed: ~150
- Code functionality: Unchanged
- Test coverage: Maintained at previous levels

## Benefits
- Improved readability of test names in code and test reports
- Better alignment with Kotlin language idioms and modern testing practices
- More descriptive test names that clearly express test intent
- Consistent naming convention across the entire codebase
- Easier maintenance and understanding of test purpose

## Related Documentation
- [Android Testing Documentation](https://developer.android.com/guide)
- [Kotlin Testing Best Practices](https://kotlinlang.org/docs/jvm-test-using-junit.html)
- [JUnit 5 Display Name Generator](https://junit.org/junit5/docs/current/user-guide/#writing-tests-display-name-generator)

### CODE REVIEW

File: app/src/test/java/org/groundplatform/android/MainViewModelTest.kt [hunk 1]
```diff
@@ -72,7 +72,7 @@ class MainViewModelTest : BaseHiltTest() {
   }
 
   @Test
-  fun testSignInStateChanged_onSignedOut() = runWithTestDispatcher {
+  fun `sign in state changed when on signed out`() = runWithTestDispatcher {
```
Comment (gino-m) [inline app/src/test/java/org/groundplatform/android/MainViewModelTest.kt]: Many of the test names are not well-formed after being renamed (e.g. "when on signed out.."). For ex. this one should be "sign in updated on sign out".

Comment (AnuragManu) [reply app/src/test/java/org/groundplatform/android/MainViewModelTest.kt]: Ack, will take the changes in.

Comment (AnuragManu) [reply app/src/test/java/org/groundplatform/android/MainViewModelTest.kt]: Changing the following 
### In MainViewModelTest.kt:
1. sign in state changed when on signed out → sign in updated on sign out
2. sign in state changed when on signing in → navigation redirects to signing in state when authentication in progress
3. sign in state changed when on signed in when tos not accepted → navigation redirects to TOS screen when signed in but terms not accepted
4. sign in state changed when on signed in get tos when tos missing → navigation redirects to TOS screen when terms document missing
5. sign in state changed when on signed in get tos when permission denied → navigation redirects to TOS screen when permission denied for terms
6. sign in state changed when on signed in get tos when not permission denied error → navigation redirects to TOS screen when error retrieving terms
7. sign in state changed when on sign in error → sign in error redirects to signed out state

### In MainActivityTest.kt:
1. signInErrorDialog_isDisplayed → signInErrorDialog_isDisplayedWhenSignInFails
2. launchAppWithSurveyId_needLogin_ShowLoginIn → launchAppWithSurveyId_whenNotLoggedIn_showsSignInScreen

**Comment by AnuragManu** [comment]
@gino-m have fixed the formatting issue, do take a look.
TY!

**Comment by AnuragManu** [comment]
Run TODO to Issue / build (pull_request) failing. looks like false error. 
The branch exist on the forked repo.
could be due to the issue 
https://github.com/google/ground-android/issues/3101.

**Comment by gino-m** [comment]
@AnuragManu It appears the checks can't check out the code from your branch. Is it public?

```
Fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +refs/heads/fix-1427*:refs/remotes/origin/fix-1427* +refs/tags/fix-1427*:refs/tags/fix-1427*
  The process '/usr/bin/git' failed with exit code 1
  Waiting 17 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +refs/heads/fix-1427*:refs/remotes/origin/fix-1427* +refs/tags/fix-1427*:refs/tags/fix-1427*
  The process '/usr/bin/git' failed with exit code 1
  Waiting 18 seconds before trying again
```

**Comment by AnuragManu** [comment]
> @AnuragManu It appears the checks can't check out the code from your branch. Is it public?
> 
> ```
> Fetching the repository
>   /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +refs/heads/fix-1427*:refs/remotes/origin/fix-1427* +refs/tags/fix-1427*:refs/tags/fix-1427*
>   The process '/usr/bin/git' failed with exit code 1
>   Waiting 17 seconds before trying again
>   /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +refs/heads/fix-1427*:refs/remotes/origin/fix-1427* +refs/tags/fix-1427*:refs/tags/fix-1427*
>   The process '/usr/bin/git' failed with exit code 1
>   Waiting 18 seconds before trying again
> ```

Yes 
Screenshot 2025-07-28 at 5 21 49 PM

**Comment by gino-m** [comment]
> > @AnuragManu It appears the checks can't check out the code from your branch. Is it public?
> > ```
> > Fetching the repository
> >   /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +refs/heads/fix-1427*:refs/remotes/origin/fix-1427* +refs/tags/fix-1427*:refs/tags/fix-1427*
> >   The process '/usr/bin/git' failed with exit code 1
> >   Waiting 17 seconds before trying again
> >   /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +refs/heads/fix-1427*:refs/remotes/origin/fix-1427* +refs/tags/fix-1427*:refs/tags/fix-1427*
> >   The process '/usr/bin/git' failed with exit code 1
> >   Waiting 18 seconds before trying again
> > ```
> 
> Yes Screenshot 2025-07-28 at 5 21 49 PM

@shobhitagarwal1612 Any ideas what could be happening here?

**Comment by shobhitagarwal1612** [comment]
It is possible that the github action is not meant to be used for forked repositories. Need to confirm this hypothesis.

**Comment by gino-m** [comment]
> It is possible that the github action is not meant to be used for forked repositories. Need to confirm this hypothesis.

Shouldn't it skip TODOs which already have a GitHub bug attached?

**Comment by shobhitagarwal1612** [comment]
The CI build has passed. So we can merge this PR of everything else looks ok. TODOs haven't been modified in this PR anyways.

**Comment by AnuragManu** [comment]
@gino-m any updates ?
do let me know if their is something i could do at my end.

**Comment by shobhitagarwal1612** [comment]
We've merged https://github.com/google/ground-android/issues/3101 which should resolve the issue with todo-to-issue workflow. Please pull latest changes from master branch

**Comment by AnuragManu** [comment]
@shobhitagarwal1612 corrected the build failure, my bad for the build failure.
